### ISSUE TITLE
(Verilog) nMigen should provide better information about original source code locations

### ISSUE THREAD

**Issue by nmigen-issue-migration** [amaranth-lang/amaranth#293]
<a href="https://github.com/Maykeye"><img src="https://avatars1.githubusercontent.com/u/13607441?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [Maykeye](https://github.com/Maykeye)**
_Saturday Jan 04, 2020 at 09:08 GMT_
_Originally opened as https://github.com/m-labs/nmigen/issues/293_

----

Consider for example this report from yosys 

        Unreached cover statement at abc.sv:2402.

If I go there, I'll see

     always @* if (\$cover$en ) cover(\$cover$check );

and I'll have to jump to `\$cover$check`, maybe even  further to variables like `\$157`.

Instead nMigen should generate 
   
     `line 123 /mnt/e/nmigen/abc.py
     always @* if (\$cover$en ) cover(\$cover$check );

In other words

a) source location should be generated next to cover/assert statements, not just next to their corresponding variable 
b) Instead of comments like ` (* src = "/mnt/e/abc/cde.py:17" *)`, source locations should be generated with \`line so tools like SymbiYosys can point the right location (maybe optionally: verilog95 doesn't describe `line directive, it was added in verilog 2001)
